internal/commands/stack/deploy_supabase.go

```diff
@@ -0,0 +1,308 @@
+package stack
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/commands"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/commands/stack/supabase"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/commands/stack/supabase/supabasechart"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/commands/stack/supabaseconfig"
+	stacksecrets "github.com/UpCloudLtd/upcloud-cli/v3/internal/commands/stack/supabaseconfig"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/output"
+	"github.com/UpCloudLtd/upcloud-go-api/v8/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/v8/upcloud/request"
+	"github.com/spf13/pflag"
+)
+
+func DeploySupabaseCommand() commands.Command {
+	return &deploySupabaseCommand{
+		BaseCommand: commands.New(
+			"supabase",
+			"Deploy a Supabase stack",
+			"upctl stack deploy supabase <project-name>",
+			"upctl stack deploy supabase my-new-project",
+		),
+	}
+}
+
+type deploySupabaseCommand struct {
+	*commands.BaseCommand
+	location   string
+	name       string
+	configPath string
+}
+
+func (s *deploySupabaseCommand) InitCommand() {
+	fs := &pflag.FlagSet{}
+	fs.StringVar(&s.location, "location", s.location, "Select the location (region) for the stack deployment")
+	fs.StringVar(&s.name, "name", s.name, "Specify the name of the Supabase project")
+	fs.StringVar(&s.configPath, "configPath", s.configPath, "Optional path to a configuration file for the Supabase stack")
+	s.AddFlags(fs)
+
+	commands.Must(s.Cobra().MarkFlagRequired("location"))
+	commands.Must(s.Cobra().MarkFlagRequired("name"))
+
+	// configPath is optional, but if provided, it should be a valid path
+	if s.configPath != "" {
+		if _, err := os.Stat(s.configPath); os.IsNotExist(err) {
+			commands.Must(s.Cobra().MarkFlagRequired("configPath"))
+		}
+	}
+}
+
+func (s *deploySupabaseCommand) ExecuteWithoutArguments(exec commands.Executor) (output.Output, error) {
+	msg := fmt.Sprintf("Creating supabase stack %v", s.name)
+	//exec.PushProgressStarted(msg)
+
+	// Command implementation for deploying a Supabase stack
+	error := deploy(s.location, s.name, s.configPath, exec)
+	if error != nil {
+		fmt.Printf("Error deploying Supabase stack: %+v\n", error)
+		return commands.HandleError(exec, msg, error)
+	}
+
+	exec.PushProgressSuccess("Supabase stack created successfully")
+
+	return output.Raw([]byte("Commamnd executed successfully")), nil
+}
+
+func deploy(location, name, configPath string, exec commands.Executor) error {
+	fmt.Printf("Deploying Supabase stack '%s' in location '%s'\n", name, location)
+
+	clusterName := fmt.Sprintf("supabase-%s-%s", name, location)
+	networkName := fmt.Sprintf("supabase-net-%s-%s", name, location)
```

Comment:
Could also include `stack-` in the name and move resource type to the end to allow using upctl all purge to remove the resources with `stack-supabase-${name}-*` pattern 🤔 

```suggestion
	clusterName := fmt.Sprintf("stack-supabase-%s-cluster", name)
	networkName := fmt.Sprintf("stack-supabase-%s-net", name)
```